cwltool/executors.py

```diff
@@ -128,35 +131,86 @@ class MultithreadedJobExecutor(JobExecutor):
     """
     Experimental multi-threaded CWL executor.
 
-    Can easily overload a system as it does not do resource accounting.
+    Does simple resource accounting, will not start a job unless it
+    has cores / ram available, but does not make any attempt to
+    optimize usage.
     """
+
     def __init__(self):  # type: () -> None
         super(MultithreadedJobExecutor, self).__init__()
         self.threads = set()  # type: Set[threading.Thread]
         self.exceptions = []  # type: List[WorkflowException]
+        self.pending_jobs = []  # type: List[JobBase]
+
+        self.max_ram = psutil.virtual_memory().total / 2**20
```

Comment:
This probably needs to be something `psutil.virtual_memory().available - PADDING` - other processes will be normally be run on a machine (e.g. OS processes) which may fluctuate their memory usage while cwltool is running.